commit 49d9faf0b11b3957d36c21f90b9f34933f507438
Author: Arnout Engelen <arnout@bzzt.net>
Date:   Sun Dec 14 10:43:57 2025 +0100

    reproducible builds: make qml dependencies explicit
    
    Similar to https://qt-project.atlassian.net/browse/QTBUG-137440
    
    To fix https://bugs.kde.org/show_bug.cgi?id=512868
    
    I'll admit I don't quite know what I'm doing here, I'm mostly guessing and mimicking, but it does seem to remove the nondeterminism :)

diff --git a/src/declarativeimports/plasmaextracomponents/CMakeLists.txt b/src/declarativeimports/plasmaextracomponents/CMakeLists.txt
index 1a7827819..a8e335af7 100644
--- a/src/declarativeimports/plasmaextracomponents/CMakeLists.txt
+++ b/src/declarativeimports/plasmaextracomponents/CMakeLists.txt
@@ -1,4 +1,8 @@
-ecm_add_qml_module(plasmaextracomponentsplugin VERSION 2.0 URI "org.kde.plasma.extras" GENERATE_PLUGIN_SOURCE)
+ecm_add_qml_module(plasmaextracomponentsplugin URI "org.kde.plasma.extras"
+  VERSION 2.0
+  GENERATE_PLUGIN_SOURCE
+  LIBRARIES org_kde_plasmacomponents3
+)
 
 target_sources(plasmaextracomponentsplugin PRIVATE
     qmenu.cpp
@@ -42,7 +46,9 @@ target_link_libraries(plasmaextracomponentsplugin PRIVATE
         Qt6::Qml
         Qt6::Widgets
         KF6::WidgetsAddons
-        Plasma::Plasma)
+        Plasma::Plasma
+        org_kde_plasmacomponents3
+)
 
 ecm_finalize_qml_module(plasmaextracomponentsplugin DESTINATION ${KDE_INSTALL_QMLDIR})
 
